packages/launch-config/src/launch-config-crud/read.ts

```diff
@@ -0,0 +1,143 @@
+                launchJson = parse(launchJsonString);
+            } else {
+                launchJsonString = await fs.readFile(launchJsonPath, { encoding: 'utf8' });
+                launchJson = parse(launchJsonString);
+            }
+            return launchJson.configurations;
+        }
+        return undefined;
+    } catch (err) {
+        throw new Error(err);
+    }
+}
+
+/**
+ * Returns all launch configurations by file from a given root folders.
+ *
+ * @param rootFolder - Single path to root or list of root folders.
+ * @param memFs - optional, the memfs editor instance.
+ * @returns {LaunchConfigInfo[]} list of launch configs.
+ */
+export async function getAllLaunchConfigs(rootFolder: string | string[], memFs?: Editor): Promise<LaunchConfigInfo[]> {
+    const launchConfigList: LaunchConfigInfo[] = [];
+    const roots = Array.isArray(rootFolder) ? rootFolder : [rootFolder];
+    const configFiles = await getLaunchJSONFilePaths(roots, memFs);
+
+    if (memFs) {
+        for (const filePath of configFiles) {
+            const launchJsonString = memFs.read(filePath);
+            const config = parse(launchJsonString) as LaunchJSON;
+
+            if (Array.isArray(config.configurations)) {
+                launchConfigList.push({ filePath, launchConfigs: config.configurations });
+            }
+        }
+    } else {
+        for (const filePath of configFiles) {
+            const launchJsonString = await fs.readFile(filePath, { encoding: 'utf8' });
+            const config = parse(launchJsonString) as LaunchJSON;
+
+            if (Array.isArray(config.configurations)) {
+                launchConfigList.push({ filePath, launchConfigs: config.configurations });
+            }
+        }
+    }
+    return launchConfigList;
+}
+
+/**
+ * Returns the launch configuration from a given launch configurations path by name.
+ *
+ * @param launchConfigPath - path to the launch.json file.
+ * @param name - name of the launch config.
+ * @param memFs - optional, the memfs editor instance.
+ * @returns {LaunchConfig} launch config.
+ */
+export async function getLaunchConfigByName(
+    launchConfigPath: string,
+    name: string,
+    memFs?: Editor
+): Promise<LaunchConfig> {
+    let launchJsonString;
+    let launchJson;
+    try {
+        if (memFs) {
+            launchJsonString = memFs.read(launchConfigPath);
+            launchJson = parse(launchJsonString);
+        } else {
+            launchJsonString = await fs.readFile(launchConfigPath, { encoding: 'utf8' });
+            launchJson = parse(launchJsonString);
+        }
+        const launchConfig = launchJson.configurations.find((c: LaunchConfig) => c.name === name);
+        if (!launchConfig) {
+            throw Error(`No config '${name}'`);
+        }
+        return launchConfig;
+    } catch (error) {
+        console.error(`Could not find launch config '${name}' in '${launchConfigPath}'`, error);
```

Comment:
We should avoid usage of `console`. We can either allow to pass in an optional logger, e.g. like done here:
https://github.com/SAP/open-ux-tools/blob/b6782872d18ca4fe459fda6ff61ac293cd4ec7e0/packages/project-access/src/project/specification.ts#L83

If we do it this way, I propose to change the signature of this function to:

```typescript
export async function getLaunchConfigByName(
    launchConfigPath: string,
    name: string,
    options?: { memFs?: Editor; logger?: Logger; }
): Promise<LaunchConfig>
```

Another option is to just throw an error with message and original error, e.g.
```typescript
throw Error(`Could not find launch config '${name}' in '${launchConfigPath}'. Error ${error}`);
```